server/store/memory/peer_store.go

```diff
@@ -280,3 +289,51 @@ func (s *peerStore) AnnouncePeers(infoHash chihaya.InfoHash, seeder bool, numWan
 
 	return
 }
+
+func (s *peerStore) GetSeeders(infoHash chihaya.InfoHash) (peers, peers6 []chihaya.Peer, err error) {
+	skey := seedersKey(infoHash)
+
+	shard := s.shards[s.shardIndex(infoHash)]
+	shard.RLock()
+	defer shard.RUnlock()
+
+	seeders := shard.peers[skey]
+	for _, p := range seeders {
+		if p.IP.To4() == nil {
+			peers6 = append(peers6, p.Peer)
+		} else {
+			peers = append(peers, p.Peer)
+		}
+	}
+	return
+}
+
+func (s *peerStore) GetLeechers(infoHash chihaya.InfoHash) (peers, peers6 []chihaya.Peer, err error) {
+	lkey := leechersKey(infoHash)
+
+	shard := s.shards[s.shardIndex(infoHash)]
+	shard.RLock()
+	defer shard.RUnlock()
+
+	leechers := shard.peers[lkey]
+	for _, p := range leechers {
+		if p.IP.To4() == nil {
+			peers6 = append(peers6, p.Peer)
+		} else {
+			peers = append(peers, p.Peer)
+		}
+	}
+	return
+}
+
+func (s *peerStore) NumSeeders(infoHash chihaya.InfoHash) (uint32, error) {
+	seeders, seeders6, err := s.GetSeeders(infoHash)
+	seeders = append(seeders, seeders6...)
```

Comment:
We really should be internally tracking these lengths rather than trying to add them up each scrape request.

Also the `Get(Seeders|Leechers)` methods should take a "numWant int" argument so that they know when to stop.